File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/addressspace/nodes/UaObjectType.java
package org.opcfoundation.webserver.addressspace.nodes;

import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.eclipse.milo.opcua.stack.core.types.builtin.NodeId;
import org.eclipse.milo.opcua.stack.core.types.enumerated.NodeClass;
import org.springframework.lang.Nullable;

public class UaObjectType extends UaDefinitionNode {
    public UaObjectType(
            NodeId nodeId,
            String browseName,
            LocalizedText displayName,
            boolean isAbstract)
    {
        super(nodeId,browseName,displayName,isAbstract);
    }

    @Override
    public NodeClass nodeClass() {
        return NodeClass.ObjectType;
    }

    public void addMember(UaInstanceNode member)
    {
        addMemberNode(member);
    }
}

File: opcua-webservice-node/src/server/addressspace/nodes/UaObjectType.ts
import { NodeClass } from "opcua-webapi";
import { UaLocalizedText, UaNodeId, VariableTypeIds } from "opcua-webapi-ts";
import { UaDefinitionNode } from "./UaDefinitionNode";
import type { UaInstanceNode } from "./UaInstanceNode";

export class UaObjectType extends UaDefinitionNode {
    constructor(
        nodeId: UaNodeId,
        browseName: string,
        displayName: UaLocalizedText,
        isAbstract: boolean) {
        super(nodeId, browseName, displayName, isAbstract);
    }

    get nodeClass(): NodeClass {
        return NodeClass.ObjectType;
    }

    addMember(member: UaInstanceNode): void {
        this.addMemberNode(member);
    }
}